feat!: revert caf.focusAllele change + add TumorVariantFrequencyStudy…

korikuzma opened the pull request:
…Result

close #319

korikuzma committed:
feat!: `CohortAlleleFrequencyStudyResult.focusAllele` can also be a `CategoricalVariant` (#318)

close #316

korikuzma committed:
feat!: revert caf.focusAllele change + add TumorVariantFrequencyStudyResult

close #319

korikuzma commented:
@kattaylor511 can't add you as a reviewer

korikuzma committed:
rm todos

korikuzma committed:
wip: add test fixture

korikuzma commented on tests/fixtures/tumor-variant-freq-study-result.yaml:
@kattaylor511 @Mrinal-Thomas-Epic ?

Mrinal-Thomas-Epic commented on tests/fixtures/tumor-variant-freq-study-result.yaml:
```diff
@@ -0,0 +1,106 @@
+    affectedTumorSamples: 17
+    totalTumorSamples: 24592
+    affectedFrequency: 0.0006912817176317501
+    sampleGroup:
+      name: cnsbrain
+      type: StudyGroup
```
I think this one shouldn't have a sample group (since we're considering all samples, just restricting the variant)

Mrinal-Thomas-Epic commented on tests/fixtures/tumor-variant-freq-study-result.yaml:
```diff
@@ -0,0 +1,106 @@
+        affectedFrequency: 0.14285714285714285
+  - type: TumorVariantFrequencyStudyResult
+    focusVariant:
+      name: CIC R215
```
Should this be CIC R215Q

korikuzma replied:
Yes, good catch. Thanks

korikuzma committed:
address pr comments

mbrush commented on tests/fixtures/tumor-variant-freq-study-result.yaml:
```diff
@@ -0,0 +1,97 @@
+          name: CIC R215W
+          type: CategoricalVariant
+        sampleGroup:
+          name: CNS/Brain
```
Consider using the `characteristics` attribute to capture a MappableConcept that formally describes the criteria restricting this sub-group.  e.g. as in the CAF Study Result example here: https://github.com/ga4gh/va-spec/blob/issue-319/tests/fixtures/cohort-allele-freq-test.yaml#L50.

Here we might say:

````
    studyGroup:
      id: remaining
      name: CNS/Brain Tumor SubGroup
      type: StudyGroup
      characteristics:
        conceptType: Disease
        name: CNS/Brain Tumors
````

And apply this change other places you define a subgroups as well . . .

mbrush commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -1013,6 +1012,61 @@ $defs:
+        - $refCurie: vrs:Allele
+        - $refCurie: gks.core:iriReference
+        - $refCurie: cat-vrs:CategoricalVariant
+      affectedTumorSamples:
```
To parallel the CAF Study Result attribute names, we had proposed `affectedSampleCount` - was there a reason for the change here?

mbrush commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -1013,6 +1012,61 @@ $defs:
+      affectedTumorSamples:
+        type: integer
+        description: The number of tumor samples that contain the focus variant
+      totalTumorSamples:
```
To parallel the CAF Study Result attribute names, we had proposed `totalSampleCount` - was there a reason for the change here?

korikuzma committed:
Merge branch '1.0.0-ballot.2025-03' into issue-319

korikuzma committed:
rename

affectedTumorSamples -> affectedSampleCount
totalTumorSamples -> totalSampleCount

korikuzma committed:
address pr comments

korikuzma commented:
@mbrush could this get a review?

ahwagner commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -1012,6 +1012,61 @@ $defs:
 
+  TumorVariantFrequencyStudyResult:
+    inherits: StudyResult
+    maturity: trial use
```
```suggestion
    maturity: draft
```

This was just ideated the other day; let's put it at draft for now and see how this plays out on a few different datasets.

ahwagner commented on tests/fixtures/tumor-variant-freq-study-result.yaml:
```diff
@@ -0,0 +1,117 @@
+  reportedIn:
+    type: Document
+    doi: 10.1158/2159-8290.CD-17-0321
+    name: Accelerating Discovery of Functional Mutant Alleles in Cancer
```
```suggestion
    title: Accelerating Discovery of Functional Mutant Alleles in Cancer
```

korikuzma committed:
update affectedFrequency

ahwagner commented on tests/fixtures/tumor-variant-freq-study-result.yaml:
```diff
@@ -0,0 +1,117 @@
+affectedFrequency: 0.001016590761223162
+focusVariant:
+  type: CategoricalVariant
+  name: CIC R215
```
Hrm. Are we not able to represent these types of variants in Cat-VRS yet? @DanielPuthawala do we have an example of a location + missense protein sequence consequence categorical variant?

korikuzma replied:
Oh, I was just keeping it simple for this example since there should be other categorical variant examples in Cat-VRS

ahwagner commented on tests/test_definitions.yaml:
```diff
@@ -88,6 +88,9 @@ tests:
   - test_file: cohort-allele-freq-test.yaml
     namespace: va-spec.base
     definition: CohortAlleleFrequencyStudyResult
+  - test_file: tumor-variant-freq-study-result.yaml
```
excellent!

ahwagner commented on tests/fixtures/tumor-variant-freq-study-result.yaml:
```diff
@@ -0,0 +1,117 @@
+            - conceptType: Disease
+              name: CNS/Brain Tumors
+        affectedSampleCount: 16
+        totalSampleCount: 17
```
This might not be correct. I would expect totalSampleCount to be the total number of samples of CNS/Brain Tumors in the full cancerhotspots dataset. This might be something that needs to be extracted from table S1 of [this publication](https://aacrjournals.org/cancerdiscovery/article/8/2/174/6249/Accelerating-Discovery-of-Functional-Mutant).

korikuzma committed:
Update schema/va-spec/base/va-core-source.yaml

Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>

korikuzma committed:
Update tests/fixtures/tumor-variant-freq-study-result.yaml

Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>

korikuzma committed:
update affectedFrequency

korikuzma committed:
Merge branch 'issue-319' of https://github.com/ga4gh/va-spec into issue-319

mbrush committed:
updates to TumorVariantFrequencyStudyResult

small clarifying changes to descriptions, sp fixes

mbrush reviewed (approved):
I made some minor changes to wording of descriptions, look good to go now!

korikuzma committed:
rerun make all

korikuzma commented:
@mbrush thanks! Just pushed changes to update the corresponding rst/json files

korikuzma committed:
rerun pre-commit
